### PR TITLE
fix(windows): comprehensive Windows compatibility overhaul

### PR DESCRIPTION
**Pull request by lEWFkRAD**
## fix(windows): platform-aware paths and bash resolution

Makes user-facing paths and shell-script execution work on Windows,
without changing POSIX behavior.

### Problem

On Windows, several user-facing strings and one code path assume POSIX:

- Prompts and CLI output hardcode `~/.hermes`, which doesn't exist on
  Windows (the real home is `~/AppData\Local\hermes`).
- `.sh`/`.bash` script execution and the launchd refresh path shell out
  to a hardcoded `/bin/bash`, which fails on Windows with a confusing
  `[WinError 2]` instead of a clear message.
- PATH/setup guidance points users at `~/.bashrc` and Linux-only flags.

### Fix

- Display the Hermes home via `display_hermes_home()` so prompts and CLI
  show the correct platform path.
- Resolve bash through `tools.environments.local._find_bash()` instead of
  hardcoded `/bin/bash` (cron `.sh` runner, gateway launchd refresh), with
  a clear error when bash is absent on Windows.
- Add Windows-specific PATH guidance in `backup`/`doctor` when the wrapper
  dir isn't on PATH, and warn that `--system`/`--run-as-user` are
  Linux-only and ignored on Windows.

### Scope

- 19 Python files, +147 / −56. No functional change on POSIX.
- LF line endings throughout; all touched files compile.
- Rebased on current `main`.

A separate docs change (normalizing `~/.hermes` → `${HERMES_HOME}` in
markdown) was intentionally **dropped** from this PR to keep it focused;
it can follow as its own PR if wanted.

### CODE REVIEW

**Comment by lEWFkRAD** [comment]
Rebased onto current main (070ac2a71) and re-verified hunk-by-hunk. Notes for reviewers:

**Dropped as already-upstream**: the `tools/environments/local.py` `_find_bash` annotation change - ad5f3341d rewrote that function entirely (Git-for-Windows preference, raises instead of returning None), so this PR no longer touches that file (19 -> 18 source files).

**Fixes folded in during the rebase** (issues in the original PR surfaced by main's churn):
- `system_prompt.py`: derive the displayed home from `get_default_hermes_root()` so profile mode doesn't render `.../profiles/coder/profiles/coder`.
- Added missing `_get_path_guidance` imports in `backup.py` / `main.py` (was a NameError).
- The new `.bat` wrapper body now carries a `:: hermes -p <profile>` marker so main's needle-based wrapper scan (d5eee133e) still detects aliases.
- PATH guidance now uses User-scope persistent `SetEnvironmentVariable` (the old Process-scope form didn't persist and the admin claim was wrong) and MSYS `/c/...` form for the bash export.
- Updated `TestCronSchedulerBashResolution` to assert the new `_find_bash` delegation instead of the removed inline lookup.

~460 tests across the touched modules pass locally; remaining local failures reproduce identically on a pristine main baseline (native-Windows environmental). Commit authorship consolidated to the AUTHOR_MAP-listed identity, which should clear the check-attribution failure.

**Comment by lEWFkRAD** [comment]
Slice-2 failure addressed in 9c32e8810: test_named_profile_line_in_prompt_text greps system_prompt.py source for the literal ~/.hermes/profiles/, which this PR deliberately replaces with a platform-aware root derived from get_default_hermes_root() (POSIX fallback literal retained). Updated the assertion to pin the new contract - the derivation call, the ~/.hermes fallback, and the /profiles/<name>/ path shape. Full test file 12/12 locally.

**Comment by AIalliAI** [comment]
Took this for a spin locally against current `main` — the change itself is solid. The `_find_bash` delegation, the `:: hermes -p <profile>` marker to keep the alias scanner matching (since the real invocation line uses `%HERMES_CMD%`, not the literal `hermes`), and the default-root fix for the `profiles/coder/profiles/coder` nesting all check out. A few things before it goes green:

**1. The failing slice-2 test is a real regression, not environmental.**
`tests/tools/test_cross_profile_guard.py:255` greps `agent/system_prompt.py`'s source for the literal `~/.hermes/profiles/`. Your f-string rewrite deletes that literal in both the default and named-profile branches, so the assert fails. I reproduced it on macOS with zero Windows involvement — it's a pure `Path(...).read_text()` source-grep, so it's deterministic and cross-platform (the `test_gateway_service.py` D-Bus failures genuinely are environmental, but this one isn't). One-line fix, same shape as what you already did for `test_windows_native_support.py`:

```python
# tests/tools/test_cross_profile_guard.py:255
-        assert "~/.hermes/profiles/" in src
+        assert "/profiles/<name>/" in src   # survives the platform-aware rewrite
```

Don't revert to the hardcoded literal though — that would reintroduce the nesting bug you just fixed.

**2. Redundant import.** `hermes_cli/main.py` re-imports `display_hermes_home` inside the `use` branch, but it's already imported at the top of `cmd_profile`. Harmless, just drop the inline one.

**3. Duplicated guidance strings.** The "add wrapper dir to PATH" POSIX text is copy-pasted across `profiles.py`, `main.py`, and `backup.py`. Since you already wrote `_get_path_guidance()` for the Windows side, might be worth routing the POSIX text through it too so the three copies can't drift.

**4. Optional hardening.** `create_wrapper_script()` interpolates `profile`/`target` into the generated `.bat` unquoted. Not exploitable today — every caller funnels through `_PROFILE_ID_RE` first — but since you're already in that function, a `validate_profile_name(profile)` at the top would keep it safe regardless of caller.

Heads up on overlap: this collides with a few earlier PRs on the same hunks — #52204 does the same `cron/scheduler.py` bash delegation (plus WSL/MSYS argv handling), and #49507 / #49974 both carry the same `display_hermes_home()` one-liner. Whoever lands first will force a rebase here.

**Comment by lEWFkRAD** [comment]
Thanks for taking it for a spin — all four items handled:

**(1) `test_cross_profile_guard.py:255`** — you were reviewing the branch as of c617efa; the follow-up commit 9c32e88 (pushed shortly after) already made exactly the change you suggest: the assert is now `"/profiles/<name>/" in src`, which survives the platform-aware rewrite without reintroducing the hardcoded root. Verified locally: the file's 12 tests pass on native Windows.

**(2) Redundant import** — dropped the inline `display_hermes_home` import in the `use` branch (0f78e9c).

**(3) Duplicated guidance strings** — `cmd_profile` and the backup-restore path now call `_get_path_guidance()` unconditionally instead of branching on platform and inlining the POSIX text; the helper already returns the right text per platform, so the three copies collapse to one (0f78e9c).

**(4) Wrapper hardening** — `create_wrapper_script()` now calls `validate_profile_name(profile)` on the target it interpolates, right beside the existing `validate_alias_name(canon)`, so the no-injection property holds regardless of caller (0f78e9c).

On the overlap warning: agreed. #52204's cron/scheduler bash delegation and the `display_hermes_home()` one-liner in #49507/#49974 touch the same hunks — happy to rebase whichever lands second; the conflicts are mechanical.

File: tests/tools/test_windows_native_support.py [hunk 1]
```diff
@@ -733,23 +733,28 @@ class TestCronSchedulerBashResolution:
...
         source = (root / "cron" / "scheduler.py").read_text(encoding="utf-8")
-        # The old hardcoded path should be gone as the sole bash source.
-        # It may still appear as a POSIX fallback after shutil.which(), so
-        # we check for the shutil.which call near the .sh/.bash branch.
-        assert 'shutil.which("bash")' in source, (
-            "cron.scheduler must resolve bash dynamically via shutil.which"
+        # cron.scheduler delegates bash resolution to the shared
+        # tools.environments.local._find_bash helper (which prefers Git
+        # for Windows bash over WSL bash and raises a clear RuntimeError
+        # when bash is missing) instead of an inline shutil.which lookup.
```
Comment (teknium1) [inline tests/tools/test_windows_native_support.py:742]: This source-string assertion does not verify that `_run_job_script()` invokes a Git Bash executable with a Windows/MSYS-safe script argument. Please add a behavioral test at the resolver/subprocess boundary; the argv is the Windows-specific failure surface.

**Review by teknium1 — COMMENTED** [review]
Thanks for consolidating the Windows path, prompt, and shell-resolution work. The underlying current-main defects remain, but the cron portion needs one more Windows-specific correction.

### Problems
- `cron/scheduler.py:2088` still invokes Git Bash with `str(path)`. The linked #52204 discussion documents native Windows paths being mangled by Git Bash/MSYS; selecting `_find_bash()` fixes executable selection but not that argv conversion.
- `tests/tools/test_windows_native_support.py:742` and `:752` only grep source text. They would not catch an incorrect script argv or a broken resolver/error integration.

### Suggested changes
- Normalize the script argument to an MSYS-safe path on Windows before building `argv`; `tools/environments/local.py:69-95` already owns that conversion logic.
- Add behavioral tests around `_run_job_script` that assert the Git Bash argv and the graceful missing-bash result.

Automated hermes-sweeper review.

File: tests/tools/test_windows_native_support.py (hunk 1 above)
Comment (lEWFkRAD) [reply tests/tools/test_windows_native_support.py:742]: Addressed in d725ed29e with test_windows_shell_script_uses_git_bash_safe_argv in tests/cron/test_cron_script.py. It drives _run_job_script under win32, stubs the shared resolver to a Git Bash executable with spaces, captures subprocess.run argv, and asserts the script argument is slash-normalized/MSYS-safe. The cron script suite passes (39 tests).

**Comment by jordan-thirkle** [comment]
## Verified: canonical test runner works on Windows 11 + Git Bash (real machine)

I ran `scripts/run_tests.sh` on a real Windows 11 / Git-Bash host (no WSL) to confirm the Windows `Scripts/` venv branch (lines 63–74) actually selects a valid interpreter:

- Env: Windows 11, Git Bash (MSYS), Python 3.11.15 via `uv` venv with pytest 9.1.1; `HERMES_PYTHON` pointed at `<venv>/Scripts/python.exe`.
- Runner correctly resolved the `Scripts/` layout (the `bin/` probe doesn't exist on native Windows venvs, so this branch is the only one that fires).
- Results: `tests/cli/test_chat_q_exit_clear.py` → 4/4 passed; `tests/tools/test_accretion_caps.py` → passed. Both green, no POSIX-only skips needed for these files.

This confirms the Windows test path is functional end-to-end on a stock Windows install. If you want CI to cover this, a minimal `windows-latest` job that does `uv venv` + `uv pip install pytest` + `scripts/run_tests.sh <small subset>` would lock it in — happy to draft that as a follow-up if useful.
